packages/clay-pagination/src/ClayPaginationWithBar.tsx

```diff
@@ -0,0 +1,184 @@
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Possible values of items per page.
+	 */
+	deltas?: Array<number>;
+
+	/**
+	 * The page numbers that should be disabled. For example, `[2,5,6]`.
+	 */
+	disabledPages?: Array<number>;
+
+	/**
+	 * The number of pages to show on each side of the active page before
+	 * using an ellipsis dropdown.
+	 */
+	ellipsisBuffer?: number;
+
+	/**
+	 * Function used to create the href provided for each page link.
+	 */
+	hrefConstructor?: (page?: number) => string;
+
+	/**
+	 * Labels for changing some texts inside the component.
+	 * Use this property for i18n.
+	 */
+	labels?: {
+		paginationResults: string;
+
+		perPageItems: string;
+
+		selectPerPageItems: string;
+	};
+
+	/**
+	 * Callback for when the number of elements per page changes. This is only used if
+	 * an href is not provided.
+	 */
+	onDeltaChange?: (page?: number) => void;
+
+	/**
+	 * Callback for when the active page changes. This is only used if
+	 * an href is not provided.
+	 */
+	onPageChange?: (page?: number) => void;
+
+	/**
+	 * Initialize the page that is currently active. The first page is `1`.
+	 */
+	initialActivePage?: number;
+
+	/**
+	 * Initializes delta. Default is `10`.
+	 */
+	initialSelectedDelta?: number;
+
+	/**
+	 * The size of pagination element.
+	 */
+	size?: 'sm' | 'lg';
+
+	/**
+	 * Path to spritemap from clay-css.
+	 */
+	spritemap: string;
+
+	/**
+	 * The total number of items in the pagination list.
+	 */
+	totalItems: number;
+}
+
+const DEFAULT_LABELS = {
+	paginationResults: 'Showing {0} to {1} of {2}',
+	perPageItems: '{0} items',
+	selectPerPageItems: '{0} items',
+};
+
+export const ClayPaginationWithBar: React.FunctionComponent<IProps> = ({
+	deltas = defaultDeltas,
+	disabledPages,
+	ellipsisBuffer,
+	hrefConstructor,
+	labels = DEFAULT_LABELS,
+	onDeltaChange = noop,
+	onPageChange = noop,
+	initialSelectedDelta = defaultDeltas[0],
+	initialActivePage = 1,
+	size,
+	spritemap,
+	totalItems,
+	...otherProps
+}: IProps) => {
+	const [active, setActive] = useState<boolean>(false);
+	const [activePage, setActivePage] = useState<number>(initialActivePage);
+	const [perPage, setPerPage] = useState<number>(initialSelectedDelta);
+
+	return (
+		<div
+			className={classNames('pagination-bar', {
+				[`pagination-${size}`]: size,
+			})}
+			{...otherProps}
+		>
+			<ClayDropDown
+				active={active}
+				className="pagination-items-per-page"
+				onActiveChange={newVal => setActive(newVal)}
+				trigger={
+					<ClayButton
+						data-testid="selectPaginationBar"
+						displayType="unstyled"
+					>
+						{sub(labels.perPageItems, [perPage])}
+						<ClayIcon
+							spritemap={spritemap}
+							symbol="caret-double-l"
+						/>
+					</ClayButton>
+				}
+			>
+				<ClayDropDown.ItemList>
+					{deltas.map((delta: number, i) => (
+						<ClayDropDown.Item
+							data-testid={`dropdownItem${i}`}
+							key={`dropdownItem${i}`}
+							onClick={() => {
```

Comment:
Maybe this should be optional, because we can use the pagination-bar with links instead of controlling the state here, doing this we are always assuming that the data will be controlled through the client side in the DXP this does not happen so often the vast majority it happens only on the server to change.

So the `deltas` API should accept an API similar to what we had in v2: https://github.com/liferay/clay/blob/2.x/packages/clay-pagination-bar/src/ClayPaginationBar.js#L107-L113

I would even choose to change the `deltas` API to `entries` seems more understanding. We must accept these two use cases.

cc @bryceosterhaus

Reply:
cc @diegonvs and @bryceosterhaus you can expose your thoughts on this here!

Reply:
I'd prefer the `deltas` name for API since its more explicit to what it is. `entries` is fairly generic. As far as onClick, I think it'd be ideal to allow either `onClick` type callbacks for more SPA focused use cases and also provide an `href` prop to account for routing use cases.

Reply:
ok, we can go with `deltas` for now so if we think we are having confusion we can change.

> As far as onClick, I think it'd be ideal to allow either onClick type callbacks for more SPA focused use cases and also provide an href prop to account for routing use cases.

Yeah, we should support both here.